src/app/events/[slug]/components/ScheduleSection/EventDetails.tsx

```diff
@@ -0,0 +1,57 @@
+'use client'
+
+import { BasicCard } from '@/components/BasicCard'
+import { Heading } from '@/components/Heading'
+import { TextLink } from '@/components/TextLink'
+
+import type { Event } from '../../../schemas/ScheduleSchema'
+import { formatTime } from '../../utils/dateUtils'
+
+import { type ParticipantsProps, ScheduleParticipants } from './Participants'
+
+export function EventDetails(event: Event) {
+  return (
+    <BasicCard>
+      <div className="grid gap-6 lg:grid-cols-3">
+        <div className="flex gap-6 text-brand-300 lg:flex-col lg:gap-1">
+          <div className="text-sm font-bold">
+            <span>{formatTime(event.start)}</span>
+            {event.end && <span> – {formatTime(event.end)}</span>}
+          </div>
+          <span className="text-sm">{event.location}</span>
+        </div>
+        <div className="lg:col-span-2">
+          <Heading tag="h3" variant="lg">
+            {event.title}
+          </Heading>
+          <div className="mt-2 space-y-2">
+            {event.description && (
+              <p className="max-w-readable">{event.description}</p>
+            )}
+            {event.moderators && event.moderators.length > 0 && (
+              <ScheduleParticipants
+                title="Moderators"
+                participants={
+                  event.moderators as ParticipantsProps['participants']
+                }
```

Comment:
If the list of participants cannot be empty, then it would be better to reflect in the type instead of using an `as` type assertion here

So here we would have:

```ts
participants={event.moderators}
```

And in the `EventSchema` we would have:

```ts
moderators: z.array(ParticipantSchema).nonempty().optional(),
```

What do you think?

Reply:
Oh, Yes! That's perfect. Thank you.